test/tools/jsontests/vm.cpp

```diff
@@ -480,114 +474,55 @@ json_spirit::mValue doVMTestsNoLog(json_spirit::mValue const& _input, bool _fill
-}
-
-BOOST_AUTO_TEST_CASE(vmBlockInfoTest)
-{
-	dev::test::executeTests("vmBlockInfoTest", "/VMTests", "/VMTestsFiller", dev::test::doVMTests);
-}
-
-BOOST_AUTO_TEST_CASE(vmIOandFlowOperationsTest)
-{
-	dev::test::executeTests("vmIOandFlowOperationsTest", "/VMTests", "/VMTestsFiller", dev::test::doVMTests);
-}
-
-BOOST_AUTO_TEST_CASE(vmPushDupSwapTest)
-{
-	dev::test::executeTests("vmPushDupSwapTest", "/VMTests", "/VMTestsFiller", dev::test::doVMTests);
-}
-
-BOOST_AUTO_TEST_CASE(vmLogTest)
-{
-	dev::test::executeTests("vmLogTest", "/VMTests", "/VMTestsFiller", dev::test::doVMTests);
-}
-
-BOOST_AUTO_TEST_CASE(vmSystemOperationsTest)
-{
-	dev::test::executeTests("vmSystemOperationsTest", "/VMTests", "/VMTestsFiller", dev::test::doVMTests);
-}
-
-BOOST_AUTO_TEST_CASE(vmPerformanceTest)
-{
-	if (test::Options::get().all)
-		dev::test::executeTests("vmPerformanceTest", "/VMTests", "/VMTestsFiller", dev::test::doVMTests);
-	else
-		cnote << "vmPerformanceTest is skipped because --all option is not specified.\n";
-}
-
-BOOST_AUTO_TEST_CASE(vmInputLimitsTest)
-{
-	if (test::Options::get().all)
-		dev::test::executeTests("vmInputLimits", "/VMTests", "/VMTestsFiller", dev::test::doVMTests);
-	else
-		cnote << "vmInputLimitsTest is skipped because --all option is not specified.\n";
-}
-
-BOOST_AUTO_TEST_CASE(vmInputLimitsLightTest)
-{
-	if (test::Options::get().all)
-		dev::test::executeTests("vmInputLimitsLight", "/VMTests", "/VMTestsFiller", dev::test::doVMTests);
-	else
-		cnote << "vmInputLimitsLightTest is skipped because --all option is not specified\n";
-}
-
-BOOST_AUTO_TEST_CASE(vmRandom)
-{
-	test::Options::get(); // parse command line options, e.g. to enable JIT
-
-	string testPath = getTestPath();
-	testPath += "/VMTests/RandomTests";
-
-	std::vector<boost::filesystem::path> testFiles = test::getJsonFiles(testPath);
-
-	test::TestOutputHelper testOutputHelper(testFiles.size());
-
-	for (auto& path: testFiles)
+public:
+	vmtestfixture()
 	{
-		try
-		{
-			cnote << "TEST " << path.filename();
-			json_spirit::mValue v;
-			string s = asString(dev::contents(path.string()));
-			BOOST_REQUIRE_MESSAGE(s.length() > 0, "Content of " + path.string() + " is empty. Have you cloned the 'tests' repo branch develop and set ETHEREUM_TEST_PATH to its path?");
-			json_spirit::read_string(s, v);
-			test::Listener::notifySuiteStarted(path.filename().string());
-			doVMTestsNoLog(v, false);
-		}
-		catch (Exception const& _e)
+		string const& casename = boost::unit_test::framework::current_test_case().p_name;
+		if (casename == "vmPerformance" && !Options::get().all)
 		{
-			BOOST_ERROR(" Failed test with Exception: " << diagnostic_information(_e));
+			cnote << "Skipping " << casename << " because --all option is not specified.\n";
+			return;
 		}
-		catch (std::exception const& _e)
+		fillAllFilesInFolder(casename);
+	}
+
+	void fillAllFilesInFolder(string const& _folder)
```

Comment:
Maybe you can make this `private`.

Reply:
I am going to remove this function in the next PR